docs/project.md

```diff
@@ -2,4 +2,154 @@
 
 A project is a an app that contains React code and has a dependency on `react-native`.
 
-Projects can provide additional properties to alter the CLI behavior (e.g. a project is located in a non-standard place) as well as change the configuration for each of the dependencies.
+Projects can provide additional properties to alter the CLI behavior, such as custom location of React Native files (this is useful when running RNTester from source) or a non-standard location of a project (useful when working in a brownfield app).
+
+## How does it work?
+
+A project can define a `react-native.config.js` at the root with custom configuration to be picked up by the CLI.
+
+For example, below configuration informs CLI of the additional assets to link and about a custom project location.
+
+```js
+module.exports = {
+  project: {
+    ios: {
+      project: './CustomProject.xcodeproj'
+    }
+  },
+  assets: ['./assets']
+};
+```
+
+You can check all available options below.
+
+## Project interface
```

Comment:
It's pretty confusing that we name it Project, while the type is User Config and contains "project" key inside, isn't it?

Reply:
Right. On the other hand, it works well with other naming patterns we have (dependency, plugin, platform).

Reply:
Renamed the type to start with.